### PR TITLE
[pytx][stacked] threatexchange match input update

### PR DESCRIPTION
**Pull request by Dcallies**
Summary
---------

This is similar to #984, except on the match command instead. I think I wrote it in a way that should be quick to port later.

Stacked on #996

Test Plan
---------

updated unittests

### CODE REVIEW

File: python-threatexchange/threatexchange/cli/helpers.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    By default assumes filename
+    If `--` is passed in, the remainder is interpreted as text
+    If empty, reads from stidin.
+
+    Use with nargs=argparse.REMAINDER for best results, which gives you:
+
+    $ echo "Some text" > file.txt; cmd file.txt
+    $ echo "Some text" | cmd
+    $ cmd -- Some text
```
Comment (sahanapeters) [inline python-threatexchange/threatexchange/cli/helpers.py:33]: 🥳

File: python-threatexchange/threatexchange/cli/tests/sample_data_e2e_test.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class SampleDataE2ETest(ThreatExchangeCLIE2eTest):
     def test_direct_to_match(self):
         """The classic first use of the CLI"""
         self.assert_cli_output(
-            ("match", "text", "-I", self.MATCHES_ONE),
+            ("match", "text", "--", self.MATCHES_ONE),
```
Comment (sahanapeters) [inline python-threatexchange/threatexchange/cli/tests/sample_data_e2e_test.py:16]: Do we not want to keep "-I" for inline?

**Review by sahanapeters — APPROVED** [review]
LGTM.

File: python-threatexchange/threatexchange/cli/tests/sample_data_e2e_test.py (hunk 2 above)
Comment (Dcallies) [reply python-threatexchange/threatexchange/cli/tests/sample_data_e2e_test.py:16]: The functionality is still there, just renamed as `--`!

**Pull request merged**